From 05cd96b1890d3bd4c6ea472edcc2e7b329e4e2e4 Mon Sep 17 00:00:00 2001
From: Allen Wild <allenwild93@gmail.com>
Date: Sun, 10 Jan 2021 14:38:49 -0500
Subject: [PATCH 06/14] cmake: allow explicitly disabling Ruby bindings

Users may wish to build Dislocker without Ruby even if it's installed on
the system. This is useful for packagers, or users who don't want to
recompile Dislocker every time their package manager updates Ruby.

Add a new WITH_RUBY cmake option that can be set to ON, OFF, or AUTO.
AUTO (the default) behaves as Dislocker does currently - check for Ruby
and use it if found, ON requires Ruby and fails if not found, and OFF
never checks for Ruby and disables the bindings.
---
 src/CMakeLists.txt | 25 +++++++++++++++++--------
 1 file changed, 17 insertions(+), 8 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index a4d2c77..df28c27 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -17,8 +17,6 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301,
 # USA.
 
-set(_Ruby_DEBUG_OUTPUT ON)
-
 if("${CMAKE_SOURCE_DIR}" MATCHES "src/?$")
 	message(FATAL_ERROR "\nPlease execute cmake from the directory above, not the src/ directory.")
 	return()
@@ -127,12 +125,23 @@ else()
 	return ()
 endif()
 
-find_package (Ruby)
-if(RUBY_FOUND  AND  RUBY_INCLUDE_DIRS  AND  RUBY_LIBRARY)
-	include_directories (${RUBY_INCLUDE_DIRS})
-	set (LIB ${LIB} ${RUBY_LIBRARY})
-	add_definitions (-D_HAVE_RUBY=${RUBY_VERSION_STRING})
-	set (SOURCES ${SOURCES} ruby.c)
+# Ruby bindings
+set(WITH_RUBY "AUTO" CACHE STRING "Enable Ruby bindings. Valid values are ON, OFF, or AUTO")
+if (NOT "${WITH_RUBY}" STREQUAL "OFF")
+	set(_Ruby_DEBUG_OUTPUT ON)
+	find_package (Ruby)
+	if(RUBY_FOUND  AND  RUBY_INCLUDE_DIRS  AND  RUBY_LIBRARY)
+		include_directories(${RUBY_INCLUDE_DIRS})
+		set(LIB ${LIB} ${RUBY_LIBRARY})
+		add_definitions(-D_HAVE_RUBY=${RUBY_VERSION_STRING})
+		set(SOURCES ${SOURCES} ruby.c)
+	elseif("${WITH_RUBY}" STREQUAL "ON")
+		message(FATAL_ERROR "Ruby bindings requested, but Ruby could not found")
+	else()
+		message("Ruby not found, Ruby bindings will be disabled")
+	endif()
+else()
+	message("Ruby bindings disabled by user request")
 endif()
 
 find_package (FUSE)
-- 
2.25.1

